storage/grpc_client.go

```diff
@@ -840,6 +841,100 @@ func (c *grpcStorageClient) DeleteHMACKey(ctx context.Context, desc *hmacKeyDesc
 	return errMethodNotSupported
 }
 
+// Notification methods.
+
+func (c *grpcStorageClient) ListNotifications(ctx context.Context, bucket string, opts ...storageOption) (n map[string]*Notification, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/storage.grpcStorageClient.ListNotifications")
+	defer func() { trace.EndSpan(ctx, err) }()
+
+	s := callSettings(c.settings, opts...)
+	if s.userProject != "" {
+		ctx = setUserProjectMetadata(ctx, s.userProject)
+	}
+	req := &storagepb.ListNotificationsRequest{
+		Parent: bucketResourceName(globalProjectAlias, bucket),
+	}
+	var notifications []*storagepb.Notification
+	err = run(ctx, func() error {
+		gitr := c.raw.ListNotifications(ctx, req, s.gax...)
+		for {
+			n, err := gitr.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			notifications = append(notifications, n)
+		}
```

Comment:
How does the JSON API behave for List Notifications? Does it get every single Notification and return it all in one blob? I think we should work around using the iterator since it complicates things relative to HTTP, WDYT?

The way we'd do that is by "iterating" ourselves via the `NotificationIterator.InteralFetch` method [here](https://github.com/googleapis/google-cloud-go/blob/main/storage/internal/apiv2/storage_client.go#L1140). We can call that until there are no more results, collecting entire pages at a time.

This could of course be even more complicated...

Reply:
IIUC, the JSON API  returns a response with `items`, which is a list of all the Notifications. 

Currently, the library returns a map of notifications instead of an iterator when we call `bucket.Notifications`, and we don't have a `type NotificationIterator`.

I tried using `NotificationIterator.InteralFetch` [here](https://github.com/googleapis/google-cloud-go/commit/09eb7a670adaf595c15c5ba60c8a411a430f8aba#diff-eec5db1002a6674758e3a557d89495986c8450de01512d561e2a7962584c08f2R861) with no success unfortunately.  Could you help point me to how to correctly use `InternalFetch` in this case?

Reply:
Ah, I would've just created an iterator via the GAPIC ListNotifications, then use _that_ iterator's InternalFetch.

Reply:
ooooh okay I think I understand, updated. what is a good value for pageSize?

Reply:
The proto documents that the max is 100. You can just omit it and it should default to something reasonable.